<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <arg name="output/cloud_topic" default="/lidar/points_preprocessed"/>
    <arg name="odom_frame" default="map"/>
    <arg name="base_frame" default="base_link"/>

    <arg name="crop_box/min_x" default="-2.0"/>
    <arg name="crop_box/max_x" default="7.0"/>
    <arg name="crop_box/min_y" default="-2.0"/>
    <arg name="crop_box/max_y" default="2.0"/>
    <arg name="crop_box/min_z" default="-400.0"/>
    <arg name="crop_box/max_z" default="500.0"/>
    <arg name="crop_box/negative" default="true"/>

    <arg name="radius_filter/radius_th" default="0.5"/>
    <arg name="radius_filter/negative" default="false"/>

    <arg name="motion_undistortion/timestamp_field_name" default="t"/>
    <arg name="motion_undistortion/velocity_threshold" default="2"/>

    <node pkg="pointcloud_preprocessing_cuda" type="pointcloud_preprocessing_cuda"
          name="pointcloud_preprocessing_cuda_node" output="screen">

        <param name="crop_box/min_x" value="$(arg crop_box/min_x)"/>
        <param name="crop_box/max_x" value="$(arg crop_box/max_x)"/>
        <param name="crop_box/min_y" value="$(arg crop_box/min_y)"/>
        <param name="crop_box/max_y" value="$(arg crop_box/max_y)"/>
        <param name="crop_box/min_z" value="$(arg crop_box/min_z)"/>
        <param name="crop_box/max_z" value="$(arg crop_box/max_z)"/>
        <param name="crop_box/negative" value="$(arg crop_box/negative)"/>

        <param name="radius_filter/radius_th" value="$(arg radius_filter/radius_th)"/>
        <param name="radius_filter/negative" value="$(arg radius_filter/negative)"/>

        <param name="motion_undistortion/timestamp_field_name" value="$(arg motion_undistortion/timestamp_field_name)"/>
        <param name="motion_undistortion/velocity_threshold" value="$(arg motion_undistortion/velocity_threshold)"/>

        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="odom_frame" value="$(arg odom_frame)"/>

        <rosparam param="input/lidars">
            [
            parent,
            front_left,
            front_right,
            back_left,
            back_right
            ]
        </rosparam>

        <remap from="out/cloud" to="$(arg output/cloud_topic)"/>

    </node>
</launch>
